test/integration/transactions/transactions.spec.test.js

```diff
@@ -119,3 +135,79 @@ describe('Transactions Spec Legacy Tests', function () {
 
   generateTopologyTests(testSuites, testContext, testFilter);
 });
+
+describe('Transactions', function () {
+  const dbName = 'retryable-handshake-tests';
+  const collName = 'coll';
+  const docs = [{ _id: 1, x: 11 }];
+  let client;
+  let db;
+  let coll;
+
+  beforeEach(function () {
+    if (
+      semver.lt(this.configuration.buildInfo.version, '4.2.0') ||
+      !VALID_TOPOLOGIES.includes(this.configuration.topologyType)
+    ) {
+      this.currentTest.skipReason =
+        'Transaction tests require MongoDB 4.2 and higher and no standalone';
+      this.skip();
+    }
+    client = this.configuration.newClient({});
+    db = client.db(dbName);
+    coll = db.collection(collName);
+  });
+
+  afterEach(async function () {
+    if (db) {
+      await db.admin().command({
+        configureFailPoint: 'failCommand',
+        mode: 'off'
+      });
+      await coll.drop();
+      await client.close();
+    }
+  });
+
+  context('when the handshake fails with a network error', function () {
```

Comment:
That ticket is a bit odd to me as it's mentioning heartbeat failures which I don't see happening in this scenario. We are failing on `saslContinue` and `ping` which are not being called from the monitor as the monitors don't auth.

Reply:
I see, I think it may be worth adding a comment anyway to both of the manual blocks flagging DRIVERS-2032 until the corresponding spec tests are unskipped in the spec files themselves, just in case we do run into issues here